storage/series.go

```diff
@@ -23,6 +23,47 @@ import (
 	"github.com/prometheus/prometheus/tsdb/tsdbutil"
 )
 
+type SeriesEntry struct {
+	Lset             labels.Labels
+	SampleIteratorFn func() chunkenc.Iterator
+}
+
+func (s *SeriesEntry) Labels() labels.Labels       { return s.Lset }
+func (s *SeriesEntry) Iterator() chunkenc.Iterator { return s.SampleIteratorFn() }
+
+type ChunkSeriesEntry struct {
+	Lset            labels.Labels
+	ChunkIteratorFn func() chunks.Iterator
+}
+
+func (s *ChunkSeriesEntry) Labels() labels.Labels     { return s.Lset }
+func (s *ChunkSeriesEntry) Iterator() chunks.Iterator { return s.ChunkIteratorFn() }
+
+// NewListSeries returns series entry with iterator that allows to iterate over provided samples.
+func NewListSeries(lset labels.Labels, s []tsdbutil.Sample) *SeriesEntry {
+	return &SeriesEntry{
+		Lset: lset,
+		SampleIteratorFn: func() chunkenc.Iterator {
+			return NewListSeriesIterator(samples(s))
+		},
+	}
+}
+
+// NewListChunkSeriesIterator returns chunk series entry that allows to iterate over provided samples.
+// NOTE: It uses inefficient chunks encoding implementation, not caring about chunk size.
+func NewListChunkSeriesFromSamples(lset labels.Labels, samples ...[]tsdbutil.Sample) *ChunkSeriesEntry {
+	return &ChunkSeriesEntry{
+		Lset: lset,
+		ChunkIteratorFn: func() chunks.Iterator {
+			chks := make([]chunks.Meta, 0, len(samples))
+			for _, s := range samples {
+				chks = append(chks, tsdbutil.ChunkFromSamples(s))
+			}
```

Comment:
1 chunk per sample? Isn't that too expensive?

Reply:
See comment: // NOTE: It uses inefficient chunks encoding implementation, not caring about chunk size.
It's used only for tests.

Reply:
That comment is missing here :) I think it was there before from where it was moved. Are we moving it here so that Thanos/Cortex can use this to write tests?

Reply:
> That comment is missing here

My bad, I see it now!